### PR TITLE
Stock Part Datumization Complete

### PR DESCRIPTION
**Pull request by SyncIt21**
So i accidently reverted all my commits in #72511 when resolving a merge conflict So ummm yeah fuck my bad anyway 

## About The Pull Request
Finishes what was started in #71693 and completes the  [initiative](https://github.com/tgstation/dev-cycles-initiative/issues/1)

Except for `obj/item/stock_parts/cell` and its subtypes. All machines now use `datum/stock_part` for its requested components & component parts

Not sure if i caught every machine & stuff in the game so merge with caution
## Changelog
:cl:
code: datum stock part for every obj stock part
refactor: all machines & dependent experiments to use datum stock parts
/:cl:

### CODE REVIEW

**Comment by SyncIt21** [comment]
@Mothblocks Sorry

**Comment by SyncIt21** [comment]
> Can you also add the following grep into the PCRE category of run_grep.sh since we're already conflicting with PRs that will use the obj paths, and it will just silently fail:
> 
> for\b.*/obj/item/stock_parts/(?!cell)(?![\w_]+ in )

I have no idea what any of this means so lets slow down for a sec

- Where is this file run_grep.sh?
- PCRE category? you mean i just copy the regex and paste it in this category inside run_grep.sh file?

**Comment by Mothblocks** [comment]
Sorry, its check_grep.sh

In there, ctrl+f `"regexes requiring PCRE2"` then add it there

**Comment by Mothblocks** [comment]
Also this is why u never force push

**Comment by SyncIt21** [comment]
> Also this is why u never force push

:D

**Comment by SyncIt21** [comment]
> Sorry, its check_grep.sh
> 
> In there, ctrl+f `"regexes requiring PCRE2"` then add it there

Found it

**Comment by SyncIt21** [comment]
```
if [ "$pcre2_support" -eq 1 ]; then
	section "regexes requiring PCRE2"
	part "to_chat sanity"
	if $grep -P 'to_chat\((?!.*,).*\)' $code_files; then
		echo
		echo -e "${RED}ERROR: to_chat() missing arguments.${NC}"
		st=1
	fi;
	... other stuff
```

So what should the part be called? what should be the echo message? Should i do something like
```
part "datum stockpart sanity"
if $grep -P 'for\b.*/obj/item/stock_parts/(?!cell)(?![\w_]+ in )' $code_files; then
		echo
		echo -e "${RED}ERROR: Should be datum/stock_part instead"
		st=1
	fi;
```

**Comment by SyncIt21** [comment]
Ok i did just that correct i guess?

**Comment by SyncIt21** [comment]
Well it worked and now i have even more stuff to fix.

File: code/game/machinery/autolathe.dm [hunk 1]
```diff
@@ -338,14 +338,14 @@
 /obj/machinery/autolathe/RefreshParts()
 	. = ..()
 	var/mat_capacity = 0
-	for(var/obj/item/stock_parts/matter_bin/new_matter_bin in component_parts)
-		mat_capacity += new_matter_bin.rating*75000
+	for(var/datum/stock_part/matter_bin/new_matter_bin in component_parts)
+		mat_capacity += new_matter_bin.tier*75000
```
Comment (jlsnow301) [inline code/game/machinery/autolathe.dm:342]:
```suggestion
		mat_capacity += new_matter_bin.tier * 75000
```

File: code/game/machinery/autolathe.dm [hunk 2]
```diff
@@ -338,14 +338,14 @@
...
 	var/datum/component/material_container/materials = GetComponent(/datum/component/material_container)
 	materials.max_amount = mat_capacity
 
 	var/efficiency=1.8
-	for(var/obj/item/stock_parts/manipulator/new_manipulator in component_parts)
-		efficiency -= new_manipulator.rating*0.2
+	for(var/datum/stock_part/manipulator/new_manipulator in component_parts)
+		efficiency -= new_manipulator.tier*0.2
```
Comment (jlsnow301) [inline code/game/machinery/autolathe.dm:348]:
```suggestion
		efficiency -= new_manipulator.tier * 0.2
```

**Review by jlsnow301 — COMMENTED** [review]
this is shaping up very nicely

File: code/game/machinery/autolathe.dm (hunk 1 above)
Comment (SyncIt21) [reply code/game/machinery/autolathe.dm:342]: How did i miss that thanks

File: code/game/machinery/autolathe.dm (hunk 2 above)
Comment (SyncIt21) [reply code/game/machinery/autolathe.dm:348]: Now i feel bad

**Comment by SyncIt21** [comment]
We have a big problem in motorized_wheelchair.dm

Unlike every other machine which stores stock parts in the `component_list` this one stores it directly in its contents i.e. it requires the physical stock part to exist in the world

https://github.com/tgstation/tgstation/blob/90385e9ac69fc6d3c8535a58364ba4f3a586065a/code/modules/vehicles/motorized_wheelchair.dm#L32-L35

And it also requires swaping the part into the users hand and moving it back into the machine

https://github.com/tgstation/tgstation/blob/90385e9ac69fc6d3c8535a58364ba4f3a586065a/code/modules/vehicles/motorized_wheelchair.dm#L96-L109

How do i fix this?

**Comment by SyncIt21** [comment]
> We have a big problem in motorized_wheelchair.dm
> 
> Unlike every other machine which stores stock parts in the `component_list` this one stores it directly in its contents i.e. it requires the physical stock part to exist in the world
> 
> https://github.com/tgstation/tgstation/blob/90385e9ac69fc6d3c8535a58364ba4f3a586065a/code/modules/vehicles/motorized_wheelchair.dm#L32-L35
> 
> And it also requires swaping the part into the users hand and moving it back into the machine
> 
> https://github.com/tgstation/tgstation/blob/90385e9ac69fc6d3c8535a58364ba4f3a586065a/code/modules/vehicles/motorized_wheelchair.dm#L96-L109
> 
> How do i fix this?

I guess the only option is to create an seperate component_list . I'll do that

**Comment by Mothblocks** [comment]
Because that's not a machine it's fine for that to just use normal machine parts, though we'll have to make sure it passes the lint...which doesn't seem to be proccing right now unless I'm mistaken?

**Comment by SyncIt21** [comment]
> Because that's not a machine it's fine for that to just use normal machine parts, though we'll have to make sure it passes the lint...which doesn't seem to be proccing right now unless I'm mistaken?

Right now i think i found a way to convert the parts to datum inside the chair. Give me 30 minutes if i give up then we can focus on the lint which is not processing now

**Comment by Mothblocks** [comment]
They don't need to be datums unless machines are involved tho

**Comment by Mothblocks** [comment]
image
Also yes this definitely should be failing, I'll investigate why later

**Comment by SyncIt21** [comment]
I've done it the motorized wheel chair is successfully datumized. Not complex at all just used the same logic for an constructable_frame i.e. delete obj stock parts & add datum stock parts to the list, & on deconstruction do the reverse process

**Comment by SyncIt21** [comment]
Also even if the speed reading says 9 upon upgrading the manipulators it still moves very slow so not sure if its a bug or intentional.

**Comment by Mothblocks** [comment]
Blegh, it feels completely pointless and more room for error, but whatever

**Comment by SyncIt21** [comment]
> Blegh, it feels completely pointless and more room for error, but whatever

:D

I have a lot of free time today

File: code/modules/food_and_drinks/machinery/smartfridge.dm [hunk 3]
```diff
@@ -458,9 +445,9 @@
 
 /obj/machinery/smartfridge/organ/RefreshParts()
 	. = ..()
-	for(var/obj/item/stock_parts/matter_bin/B in component_parts)
-		max_n_of_items = 20 * B.rating
-		repair_rate = max(0, STANDARD_ORGAN_HEALING * (B.rating - 1) * 0.5)
+	for(var/datum/stock_part/matter_bin/B in component_parts)
+		max_n_of_items = 20 * B.tier
+		repair_rate = max(0, STANDARD_ORGAN_HEALING * (B.tier - 1) * 0.5)
```
Comment (jlsnow301) [inline code/modules/food_and_drinks/machinery/smartfridge.dm:450]:
```suggestion
	for(var/datum/stock_part/matter_bin/bin in component_parts)
		max_n_of_items = 20 * bin.tier
		repair_rate = max(0, STANDARD_ORGAN_HEALING * (bin.tier - 1) * 0.5)
```

File: code/modules/wiremod/core/component_printer.dm [hunk 4]
```diff
@@ -350,16 +350,16 @@
...
 
 		materials.set_local_size(total_storage)
 
 	var/rating = 0
 
-	for(var/obj/item/stock_parts/manipulator/manipulator in component_parts)
+	for(var/datum/stock_part/manipulator/manipulator in component_parts)
 		///we do -1 because normal manipulators rating of 1 gives us 1-1=0 i.e no decrement in cost
-		rating += manipulator.rating-1
+		rating += manipulator.tier-1
```
Comment (jlsnow301) [inline code/modules/wiremod/core/component_printer.dm:362]:
```suggestion
		rating += manipulator.tier - 1
```

**Comment by Mothblocks** [comment]
Can you please push a commit that reverts one of the loops back to the old one so we can make sure the lint works?

**Comment by SyncIt21** [comment]
> Can you please push a commit that reverts one of the loops back to the old one so we can make sure the lint works?

Which looks specifically

**Comment by Mothblocks** [comment]
Literally any I just want to see if it works

**Comment by SyncIt21** [comment]
Testing
```
/obj/machinery/smartfridge/organ/RefreshParts()
	. = ..()
	for(var/obj/item/stock_parts/matter_bin/bin in component_parts)
		max_n_of_items = 20 * bin.rating
		repair_rate = max(0, STANDARD_ORGAN_HEALING * (bin.rating - 1) * 0.5)
```

**Comment by SyncIt21** [comment]
Also you should add the "Test Merge Candidate" label like the previous one

**Comment by Mothblocks** [comment]
cool

**Comment by Mothblocks** [comment]
Ugh, didn't notice but you did not follow my review of renaming the names to tier1, tier2, etc. That was not a nitpick I am sick of the names in our codebase being useless, please fix that in a later PR

**Comment by SyncIt21** [comment]
> Ugh, didn't notice but you did not follow my review of renaming the names to tier1, tier2, etc. That was not a nitpick I am sick of the names in our codebase being useless, please fix that in a later PR

To be continued....

**Pull request merged**